diff --git a/block_townsquare.php b/block_townsquare.php
index ff33627..f5cfeb6 100644
--- a/block_townsquare.php
+++ b/block_townsquare.php
@@ -49,10 +49,11 @@ class block_townsquare extends block_base {
         $townsquarecontent = $controller->build_content();
         $mustachedata = new stdClass();
         $mustachedata->content = $townsquarecontent;
-        //$townsquareevents = $controller->townsquareevents->townsquare_get_calendarevents();
+        ob_start();
+        var_dump($controller->townsquareevents->townsquare_get_calendarevents());
         $this->content = new stdClass;
-        $this->content->text = $OUTPUT->render_from_template('block_townsquare/main', $mustachedata);
-
+        //$this->content->text = $OUTPUT->render_from_template('block_townsquare/main', $mustachedata);
+        $this->content->text = ob_get_clean();
         $this->page->requires->js_call_amd('block_townsquare/postletter', 'init');
         return $this->content;
     }
diff --git a/classes/contentcontroller.php b/classes/contentcontroller.php
index 9a23b77..e348f6d 100644
--- a/classes/contentcontroller.php
+++ b/classes/contentcontroller.php
@@ -58,22 +58,13 @@ class contentcontroller {
 
     // Core functions.
 
-    /**
-     * Uses the townsquareevents class to retrieve all important events.
-     * @return array
-     */
-    public function retrieve_events() {
-        $this->events = $this->townsquareevents->townsquare_get_all_events_sorted();
-        return $this->events;
-    }
-
     /**
      * Builds the content from events.
      * @return array
      * @throws moodle_exception
      */
     public function build_content() {
-        $this->retrieve_events();
+        $this->events = $this->townsquareevents->townsquare_get_all_events_sorted();
 
         $orientationmarkerset = false;
         $index = 0;
diff --git a/classes/townsquareevents.php b/classes/townsquareevents.php
index 23b558d..c25d259 100644
--- a/classes/townsquareevents.php
+++ b/classes/townsquareevents.php
@@ -102,7 +102,7 @@ class townsquareevents {
     public function townsquare_get_calendarevents() : array {
         global $DB, $USER;
         // Get all events from the last six months and the next six months.
-        //$calendarevents = calendar_get_events($this->starttime, $this->endtime, true, true, $this->courses, true, true, false);
+
         $calendarevents = $this->townsquare_search_events($this->timestart, $this->timeend, $this->courses);
 
         // Filter the events and add the coursemoduleid.
@@ -110,8 +110,15 @@ class townsquareevents {
             $calendarevent->coursemoduleid = get_coursemodule_from_instance($calendarevent->modulename, $calendarevent->instance,
                                                                             $calendarevent->courseid)->id;
 
-            // If there is an assignment event, check if it is relevant for the current user.
+            // Delete assign events that the user should not see.
             if ($calendarevent->modulename == "assign") {
+
+                // If the assignment due date is over than a week, it disappears.
+                if ($calendarevent->eventtype == "due" && $this->timenow >= ($calendarevent->timestart + 604800)) {
+                    unset($calendarevents[$calendarevent->id]);
+                    continue;
+                }
+
                 // Only people that can rate should see a gradingdue event.
                 if ($calendarevent->eventtype == "gradingdue" &&
                     !has_capability('mod/assign:grade', \context_module::instance($calendarevent->coursemoduleid))) {
@@ -126,13 +133,8 @@ class townsquareevents {
                     unset($calendarevents[$calendarevent->id]);
                     continue;
                 }
-
-                // If the assignment due date is over than a week, it disappears.
-                if ($this->timenow >= ($assignment->duedate + 604800)) {
-                    unset($calendarevents[$calendarevent->id]);
-                    continue;
-                }
             }
+
             // Delete activity completions that are completed by the current user.
             if ($calendarevent->eventtype == "expectcompletionon") {
                 if ($completionstatus = $DB->get_record('course_modules_completion',
@@ -268,9 +270,9 @@ class townsquareevents {
     /**
      * Searches for events in the events table, that are relevant to the timeline.
      * This is a helper function for townsquare_get_calendarevents().
-     * @param $timestart
-     * @param $timeend
-     * @param $courses
+     * @param int $timestart The time from where the events should be searched. Not equal to timestart in the database events table.
+     * @param int $timeend   The time until where the events should be searched.
+     * @param array $courses The ids of the courses where the events should be searched.
      * @return array
      * @throws dml_exception
      */
